File: data_structures/Stack/swift/Stack.swift
struct Stack<Element> {
    
    fileprivate var array: [Element] = []
    
    // Show top (last) element
    func peek() -> Element? {
        return array.last;
    }
    
    // Add an element to the stack
    mutating func push(_ element: Element) {
        array.append(element)
    }
    
    // Remove top (last) element from the stack
    mutating func pop() -> Element? {
        return array.popLast()
    }
    
    // Return number of elements in the stack
    func count() -> Int {
        return array.count
    }
    
    // Returns true when the stack is empty
    func isEmpty() -> Bool {
        return array.isEmpty
    }
    
}

File: data_structures/Stack/rust/stack.rs
struct Stack<T> {
    data: Vec<T>,
}

impl<T> Stack<T> {
    //  Returns a new Stack
    fn new() -> Stack<T> {
        Stack { data: vec![] }
    }

    // Pushes the elemnt to top of stack
    fn push(&mut self, e: T) {
        self.data.push(e)
    }

    // Returns Option with top element of the stack
    fn pop(&mut self) -> Option<T> {
        self.data.pop()
    }

    // Returns true if stack is empty
    fn is_empty(&self) -> bool {
        self.data.is_empty()
    }

    // Returns length of Stack
    fn len(&self) -> usize {
        self.data.len()
    }
}
